internal/cli/capact/parameters.go

```diff
@@ -160,6 +165,38 @@ func (i *InputParameters) ResolveVersion() error {
 			i.Override.CapactValues.Global.ContainerRegistry.Tag = LocalDockerTag
 		}
 	}
+
+	// if not already set via flags, resolve base on our logic
+	if i.ActionCRDLocation == "" {
+		if err := i.resolveCRDLocationFromVersion(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// resolveCRDLocationFromVersion sets the CRD location.
+// If version was:
+// - local tag, use the relative local CRD path
+// - stable release (tag), use tag
+// - the latest release from main (tag-commit), use the commit sha
+func (i *InputParameters) resolveCRDLocationFromVersion() error {
+	if i.Version == LocalVersionTag {
+		i.ActionCRDLocation = LocalCRDPath
+		return nil
+	}
+
+	decoded, err := semver.NewVersion(i.Version)
+	if err != nil {
+		return err
+	}
+	if decoded.Prerelease() != "" { // version in format {tag-commit}
+		i.ActionCRDLocation = fmt.Sprintf(CRDUrlFormat, decoded.Prerelease())
+	} else { // version in format {tag}
+		ghTag := fmt.Sprintf("v%s", decoded.String())
+		i.ActionCRDLocation = fmt.Sprintf(CRDUrlFormat, ghTag)
+	}
```

Comment:
```suggestion

	if decoded.Prerelease() != "" { // version in format {tag-commit}
		i.ActionCRDLocation = fmt.Sprintf(CRDUrlFormat, decoded.Prerelease())
		return nil
	}
	
	// version in format {tag}
	ghTag := fmt.Sprintf("v%s", decoded.String())
	i.ActionCRDLocation = fmt.Sprintf(CRDUrlFormat, ghTag)
```

Reply:
discussed f2f, I will leave it as it is, as it is a single decision block, when we will split that IMO we will lose the relation between it.